File: TsinghuaNet.Avalonia/Commands/DropCommand.vb
Imports System.Windows.Input

Class DropCommand
    Implements ICommand

    Public Sub Execute(parameter As Object) Implements ICommand.Execute
        Throw New NotImplementedException()
    End Sub

    Public Event CanExecuteChanged As EventHandler Implements ICommand.CanExecuteChanged

    Public Function CanExecute(parameter As Object) As Boolean Implements ICommand.CanExecute
        Return True
    End Function
End Class

File: TsinghuaNet.Uno/TsinghuaNet.Uno.Shared/Models/DropCommand.cs
using System;
using System.Net;
using System.Windows.Input;

namespace TsinghuaNet.Uno.Models
{
    sealed class DropCommand : ICommand
    {
#pragma warning disable 0067
        public event EventHandler CanExecuteChanged;
#pragma warning restore 0067

        public event EventHandler<IPAddress> DropUser;

        public void Execute(object parameter) => DropUser?.Invoke(this, (IPAddress)parameter);

        public bool CanExecute(object parameter) => parameter != null;
    }
}
